src/styles/main.scss

```
@@ -1,3 +1,92 @@
+$days:
+  "tue" 1,
+  "wed" 2,
+  "thu" 3,
+  "fri" 4,
+  "sat" 5,
+  "sun" 6;
+
+%container {
+  background-color: #eee;
+  height: 100px;
+  width: 100px;
+  border: 1px solid black;
+}
+
+* {
+  box-sizing: border-box;
+  font-family: inherit;
+}
+
+html {
+  font-family: Arial, Helvetica, sans-serif;
+  font-size: 30px;
+}
+
 body {
   margin: 0;
+  display: flex;
+  justify-content: center;
+}
+
+.calendar {
+  width: 750px;
+  margin-top: 50px;
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1px;
+  div {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    &:hover {
+      margin-top: -20px;
+      background-color: pink;
+      transition: 500ms ease;
+      cursor: pointer;
+    }
+  }
+
+}
+
+@each $day,$i in $days {
+  .calendar-#{$day} div:first-child {
+    margin-left: 101px*$i;
+  }
+}
+
+@for $i from 1 through 28 {
+  .calendar div:nth-child(#{$i}) {
+    @extend %container;
+    &::before {
+      content: "#{$i}";
+    }
+  }
+}
+
+@for $i from 29 to 30 {
+  .calendar-29 div:nth-child(#{$i}) {
+    @extend %container;
+    &::before {
+      content: "#{$i}";
+    }
+  }
+}
+
+@for $i from 29 to 31 {
+  .calendar-30 div:nth-child(#{$i}) {
+    @extend %container;
+    &::before {
+      content: "#{$i}";
+    }
+  }
+}
+
+@for $i from 29 to 32 {
+  .calendar-31 div:nth-child(#{$i}) {
+    @extend %container;
+    &::before {
+      content: "#{$i}";
```

Ці правила можна задати одним циклом

---

Вдалося скоротити до двох циклів, як дороблю сьогоднішній таск, спробую покращити, а поки що так...
```suggestion
@for $i from 1 through 31 {
  .calendar div:nth-child(#{$i}) {
    &::before {
      content: "#{$i}";
    }
  }
}

@for $i from 28 through 31 {
  @for $j from $i through 31 {
    .calendar-#{$i} div:nth-child(#{$j+1}) {
      display: none;
    }
  }
}
```